tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/PublicClientApplicationTests.cs

```diff
@@ -1002,25 +1003,96 @@ public void B2CAcquireTokenWithValidateAuthorityTrueAndRandomAuthorityTest()
         [TestMethod]
         [TestCategory("PublicClientApplicationTests")]
         public void B2CAcquireTokenWithB2CLoginAuthorityTest()
+        {
+            using (var harness = new MockHttpAndServiceBundle())
+            {
+                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthority, false);
+                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityBlackforest, false);
+                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityMoonCake, false);
+                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityUsGov, false);
+                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CCustomDomain, false);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("PublicClientApplicationTests")]
+        public void B2CAcquireTokenWithMicrosoftLoginAuthorityTest()
         {
             using (var harness = new MockHttpAndServiceBundle())
             {
                 harness.HttpManager.AddInstanceDiscoveryMockHandler();
 
-                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CAuthority);
-                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthority);
-                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityBlackforest);
-                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityMoonCake);
-                ValidateB2CLoginAuthority(harness, MsalTestConstants.B2CLoginAuthorityUsGov);
+                var app = PublicClientApplicationBuilder
+                   .Create(MsalTestConstants.ClientId)
+                   .WithB2CAuthority(MsalTestConstants.B2CAuthority, true)
+                   .WithHttpManager(harness.HttpManager)
+                   .BuildConcrete();
+
+                var ui = new MockWebUI()
+                {
+                    MockResult = new AuthorizationResult(
+                        AuthorizationStatus.Success,
+                        MsalTestConstants.B2CAuthority + "?code=some-code")
+                };
+
+                MsalMockHelpers.ConfigureMockWebUI(app.ServiceBundle.PlatformProxy, ui);
+                harness.HttpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.B2CAuthority);
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(MsalTestConstants.B2CAuthority);
+
+                var result = app
+                    .AcquireTokenInteractive(MsalTestConstants.Scope, null)
+                    .ExecuteAsync(CancellationToken.None)
+                    .Result;
+
+                Assert.IsNotNull(result);
+                Assert.IsNotNull(result.Account);
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("PublicClientApplicationTests")]
```

Comment:
I don't see any advantage of having a Test category that is the same as the name of the class.